mongomock/database.py

```diff
@@ -63,10 +65,39 @@ def collection_names(self, include_system_collections=True, session=None):
 
         return self.list_collection_names(session=session)
 
-    def list_collection_names(self, session=None):
+    def list_collection_names(self, filter=None, session=None):
+        """filter: only name field type with eq,ne or regex operator
+
+        session: not supported
+        supported operator are $eq, $ne, $regex
+        """
+
+        allowed_operators = ['$regex', '$eq', '$ne']
+        filed_name = 'name'
+
+        def verify_supported_op(keys):
+            for key in keys:
```

Comment:
Note that if you convert allowed_operators to a set, and make sure to use a set for keys, you can check directly `if keys - allowed_operators`.

Reply:
nice